project/backend/src/main/java/com/google/sps/servlets/GenerateSwipeMatchGroupIdServlet.java

```diff
@@ -0,0 +1,89 @@
+package com.google.sps.servlets;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/swipe-match/generate-group-id")
+public class GenerateSwipeMatchGroupIdServlet extends HttpServlet {
+
+  private static final String PROJECT_ID = "mak-step-2020";
+  private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  private static final String NUMBERS = "0123456789";
+  private static final int GROUP_ID_LENGTH = 8;
+  private static final String COLLECTION_NAME = "swipe-match-sessions";
+  private static final int RETRY_LIMIT = 10;
+  
+  private FirestoreOptions firestoreOptions;
+  private Firestore db;
+
+  private class RetryLimitExceededException extends Exception {
+    public RetryLimitExceededException(String message) {
+      super(message);
+    }
+  }
+
+  @Override
+  public void init() {
+    try {
+      firestoreOptions = FirestoreOptions.getDefaultInstance().toBuilder()
+        .setProjectId(PROJECT_ID)
+        .setCredentials(GoogleCredentials.getApplicationDefault())
+        .build();
+      
+      db = firestoreOptions.getService();
+    } catch (IOException e) {
+      // Not sure what to do if initialization fails...
```

Comment:
hmmm, this would mean we can't use the db for handling requests, so we could make it Optional<Firestore> (https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html ) db and if we hit an IOException, initialize it to Optional.empty -- then on doGet -- we can send an internal server error in the response if the db doesn't exist